src/libs/recentcollect.c

```diff
@@ -274,23 +331,35 @@ static void _lib_recentcollection_updated(gpointer instance, dt_collection_chang
 #endif
       gtk_label_set_ellipsize(GTK_LABEL(child), PANGO_ELLIPSIZE_END);
     }
-    gtk_widget_set_no_show_all(d->item[k].button, TRUE);
-    gtk_widget_set_visible(d->item[k].button, FALSE);
+    gtk_widget_set_no_show_all(d->items[k].button, TRUE);
+    gtk_widget_set_visible(d->items[k].button, FALSE);
   }
-  for(int k = 0; k < CLAMPS(dt_conf_get_int("plugins/lighttable/recentcollect/num_items"), 0, NUM_LINES); k++)
+  for(int k = 0; k < CLAMPS(dt_conf_get_int("plugins/lighttable/recentcollect/num_items"), 0, d->nb_items); k++)
   {
-    gtk_widget_set_no_show_all(d->item[k].button, FALSE);
-    gtk_widget_set_visible(d->item[k].button, TRUE);
+    gtk_widget_set_no_show_all(d->items[k].button, FALSE);
+    gtk_widget_set_visible(d->items[k].button, TRUE);
   }
 
   dt_thumbtable_set_offset(table, new_pos, TRUE);
 }
 
+void _init_recentcollect_max_items()
+{
+  recentcollect_max_items = dt_conf_get_int("plugins/lighttable/recentcollect/max_items");
+
+  if (!recentcollect_max_items)
```

Comment:
In fact, if max_items is not yet set, dt_conf_get_int() returns the default value. 
Then DEFAULT_MAX_ITEMS and this test are not needed.